### PR TITLE
76 implement agency maintenance pages

### PR DESCRIPTION
**Pull request by tom-karasch**
Pages done. Working on unit tests.  WIP

### CODE REVIEW

**Review by dougwaldron — COMMENTED** [review]
| Device | URL |
|--|--|
| desktop | https://localhost:7229 |

    
| Device | URL |
|--|--|
| mobile | https://localhost:7229 |

Not what you expected? Are your scores flaky? [Try running on Foo instead](https://github.com/foo-software/lighthouse-check#foos-automated-lighthouse-check-api-usage)

**Comment by tom-karasch** [comment]
Two tests still not working correctly, but feel free to review, Doug

File: src/AppServices/Agencies/IAgencyService.cs [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+using Sbeap.AppServices.ActionItemTypes;
+
 namespace Sbeap.AppServices.Agencies;
 
 public interface IAgencyService : IDisposable
 {
-    Task<IReadOnlyList<ListItem>> GetListItemsAsync(bool activeOnly = true, CancellationToken token = default);
+    Task<AgencyViewDto?> FindAsync(Guid id, CancellationToken token = default);
+    Task<IReadOnlyList<AgencyViewDto>> GetListItemsAsync(CancellationToken token = default);
+    Task<IReadOnlyList<ListItem>> GetActiveListItemsAsync(CancellationToken token = default);
```
Comment (dougwaldron) [inline src/AppServices/Agencies/IAgencyService.cs:10]: You still need a method to return all Agencies as list items, not just active ones. (The search form needs to be able to search for inactive Agencies as well as active.)

**Review by dougwaldron — CHANGES_REQUESTED** [review]
Please make the following changes. They are mostly similar to the ones from my Action Item Type review.

Be careful, there's a difference between `GetList`, which returns a list of DTO items, and `GetListItems`, which returns a list of the `ListItem` type. I think that's what was causing the problem on the Case index and edit pages.

File: src/AppServices/Agencies/Permissions/AgencyOperation.cs
[the remark this answers is not in the document]
Comment (tom-karasch) [reply src/AppServices/Agencies/Permissions/AgencyOperation.cs:1]: See my comment from Action Item Types about Permissions for Site Maintenance. If no permissions are used, any user ca add or edit drop down items. Unless the line: 
**[Authorize(Policy = PolicyName.SiteMaintainer)]**
is all that is necessary

File: src/WebApp/Pages/Cases/Index.cshtml.cs [hunk 2]
```diff
@@ -65,7 +65,7 @@ public async Task<IActionResult> OnGetSearchAsync(CaseworkSearchDto spec, [FromQ
     }
 
     private async Task PopulateSelectListsAsync() =>
-        AgencySelectList = (await _agencyService.GetListItemsAsync(false)).ToSelectList();
+        AgencySelectList = (SelectList)await _agencyService.GetActiveListItemsAsync();
```
[the remark this answers is not in the document]
Comment (tom-karasch) [reply src/WebApp/Pages/Cases/Index.cshtml.cs]: Doug, I am still not able to fix this. I don't understand what is going on here. I made changes that I thought were correct, but still is not working.

**Comment by tom-karasch** [comment]
Doug,
I think I made all necessary changes including modifying the CaseworkUpdatePermissionsHandler. Not sure if this is correct, but it works now. Also, just wanted to know about my questions: https://github.com/gaepdit/sbeap/pull/111#discussion_r1267336728
https://github.com/gaepdit/sbeap/pull/109#discussion_r1267287599

Are the permissions to edit/add/activate ActionItemTypes and Agencies controlled by the Razor Page syntax ---> 
**[Authorize(Policy = PolicyName.SiteMaintainer)]**
Does this make the AppServices Permissions for ActionItemTypes and Agencies unnecessary?? 

BTW, This pull request will also cover PR#109. So, you should be able to close both by merging this PR.

**Comment by tom-karasch** [comment]
Made requested changes

**Comment by dougwaldron** [comment]
SonarCloud analysis failure is due to code duplication between action item types and agencies, which is acceptable.

**Pull request merged**